concordium-std-derive/src/lib.rs

```diff
@@ -2121,6 +2125,66 @@ fn concordium_test_worker(_attr: TokenStream, item: TokenStream) -> syn::Result<
     Ok(test_fn.into())
 }
 
+#[cfg(feature = "concordium-quickcheck")]
+#[proc_macro_attribute]
+/// Derive the appropriate export for an annotated quickcheck function, when
+/// feature "wasm-test" is enabled, otherwise behaves like `#[quickcheck]`.
+pub fn concordium_quickcheck(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    let output = match syn::Item::parse.parse(input.clone()) {
+        Ok(syn::Item::Fn(mut item_fn)) => {
+            let mut inputs = syn::punctuated::Punctuated::new();
+            let mut errors = Vec::new();
+
+            item_fn.sig.inputs.iter().for_each(|input| match *input {
+                syn::FnArg::Typed(syn::PatType {
+                    ref ty,
+                    ..
+                }) => {
+                    inputs.push(parse_quote!(_: #ty));
+                }
+                _ => errors.push(syn::parse::Error::new(
+                    input.span(),
+                    "unsupported kind of function argument",
+                )),
+            });
+
+            if errors.is_empty() {
+                let attrs = mem::take(&mut item_fn.attrs);
+                let name = &item_fn.sig.ident;
+                let fn_type = syn::TypeBareFn {
+                    lifetimes: None,
+                    unsafety: item_fn.sig.unsafety,
+                    abi: item_fn.sig.abi.clone(),
+                    fn_token: <syn::Token![fn]>::default(),
+                    paren_token: syn::token::Paren::default(),
+                    inputs,
+                    variadic: item_fn.sig.variadic.clone(),
+                    output: item_fn.sig.output.clone(),
+                };
```

Comment:
Is it not possible to use `quote` instead of this? If not, can you at least elaborate what the purpose here is?

Reply:
This code was mostly copied from the QuckCheck macro. Here, it uses the type of the annotated function to construct the type of the function to call `concordium_std::test_infrastructure::concordium_qc`. Is there a way to do this better?

Reply:
See the `quote!` macro we use already in this file.
It would make the code more readable because it allows you to write the code using rust syntax

Reply:
Fixed. Now it looks much simpler.